brave-instrumentation/aws-java-sdk-core/src/test/java/brave/instrumentation/aws/FakeSyncClient.java

```diff
@@ -0,0 +1,76 @@
+package brave.instrumentation.aws;
+
+import com.amazonaws.AmazonWebServiceClient;
+import com.amazonaws.AmazonWebServiceRequest;
+import com.amazonaws.ClientConfigurationFactory;
+import com.amazonaws.DefaultRequest;
+import com.amazonaws.Request;
+import com.amazonaws.client.AwsSyncClientParams;
+import com.amazonaws.client.builder.AwsSyncClientBuilder;
+import com.amazonaws.http.HttpMethodName;
+import com.amazonaws.http.HttpResponse;
+import com.amazonaws.http.HttpResponseHandler;
+import java.io.ByteArrayInputStream;
+import java.net.URI;
+
+class FakeSyncClient extends AmazonWebServiceClient {
```

Comment:
@devinsba @llinder so this probably needs to be done also for the async side of things. took a bit of trail-error, but I think it is a decent approximation of how any given SDK client works.